### ISSUE TITLE
Voting

### ISSUE THREAD

**Issue by puddingspudding** [puddingspudding/persistent-video-volume#6]
To further improve the user's experience there is a decision to make.
I'm not sure if a global audio volume (as it is right now) or a per website based volume should be implemeted.

Please vote using +1 in  the comments below or suggest any other idea.

Thanks

**Comment by puddingspudding** [comment]
Keep global audio volume

**Comment by puddingspudding** [comment]
Change to audio volume per website

**Comment by puddingspudding** [comment]
Hi @ThomasDecavele , @drakray and @GOKUSSL4 ,
Since you are contributing to this project in any kind, I would like to ask you for your opinion.

**Comment by harryray33** [comment]
I've just come across this addon...although it's a global setting across all sites the addon and site volume are independent of each other.
This addon has the same problem though of the volume increasing to maximum on a browser restart.

**Comment by puddingspudding** [comment]
@harryray33 Lets start again to get the problem clear stated :)
Which browser do you use?
Which operating system do you use?
Does this always happen?
Does this affect every website?
Do you use an incognito tab/window?

**Comment by harryray33** [comment]
Firefox v. 55.0.3 64 bit
Win7
Yes, happens every time
Not sure if it happens on every site as at present I'm just using it on Dailymotion.
I don't use incognito window

**Comment by puddingspudding** [comment]
I think I'm going to implement to following:
2 Controls. First one is the max global volume. Its only applied if the websites volume is higher and any change to the websites volume wont be reset if changed below this value.

Second one is a  per website. Any change to this control will always affect the website volume. And any change to the websites "native" volume control changes the addons value.
The rule of the global max control wont apply if the value of the second control is higher.

Basically changes to current addon:
- Change the current control to be websites based.
- Add max volume control (only applies volume change to the website if volume is above)

I think, this covers most of the use cases

**Comment by harryray33** [comment]
Sorry, I'm being thick!! Does that mean that if I change the  website volume it wont alter the add-on volume?

Did you manage to work out the problem of the volume defaulting to 100% on browser restart? For me, it's one of the times that I need to control the volume the most.

Thanks.

**Comment by puddingspudding** [comment]
> Does that mean that if I change the website volume it wont alter the add-on volume?

Yes and No :). One of the controls will always be in sync with the website volume. If not, the addon will set the websites volume. Any change in the addon or website will affect the other one.

The other volume control in the addon wont be affect at all. It only serves as an upper limit and is only used if the websites volume if above



> Did you manage to work out the problem of the volume defaulting to 100% on browser restart? For me, it's one of the times that I need to control the volume the most

I haven't looked into the Dailymotion problem yet

**Comment by harryray33** [comment]
I'll try it when you release the new version...
Would it be possible maybe to have an option to either have the add-on volume connected or disconnected from the site volume?

I've tried a browser restart many times now and consistently getting the same problem with the volume. I'll give a couple of other sites a try as well such as Vimeo and Vidzi.

**Comment by GOKUSSL4** [comment]
oh yeah my main browser is firefox quantum.
Windows 10 up to date.
The issue i had it would take over volume control of a website which made any volume adjustments in the webpage not work. The addon would work ,but the website controls had no effect.

**Comment by harryray33** [comment]
New version is freezing my Firefox (FF v.55.0.3) Have to restart browser
